recovery_test.go

```diff
@@ -568,27 +586,33 @@ func TestRecoverFromMultipleNotarizations(t *testing.T) {
 	require.NoError(t, err)
 	wal.Append(firstNotarizationRecord)
 
-	protocolMetadata.Round = 2
+	protocolMetadata.Round = 1
+	protocolMetadata.Seq = 1
 	secondBlock, ok := bb.BuildBlock(ctx, protocolMetadata)
 	require.True(t, ok)
 	record = BlockRecord(secondBlock.BlockHeader(), secondBlock.Bytes())
 	wal.Append(record)
 
-	// Add notarization for second block
-	secondNotarizationRecord, err := newNotarizationRecord(l, sigAggregrator, secondBlock, nodes[0:quorum])
-	require.NoError(t, err)
-	wal.Append(secondNotarizationRecord)
+	protocolMetadata.Round = 2
+	protocolMetadata.Seq = 2
+	thirdBlock, ok := bb.BuildBlock(ctx, protocolMetadata)
+	require.True(t, ok)
+	record = BlockRecord(thirdBlock.BlockHeader(), thirdBlock.Bytes())
+	wal.Append(record)
 
-	// Create finalization record for first block
-	_, finalizationRecord := newFinalizationRecord(t, l, sigAggregrator, firstBlock, nodes[0:quorum])
-	wal.Append(finalizationRecord)
+	fCert1, _ := newFinalizationRecord(t, l, sigAggregrator, firstBlock, nodes[0:quorum])
+	fCert2, _ := newFinalizationRecord(t, l, sigAggregrator, secondBlock, nodes[0:quorum])
+	fCer3, _ := newFinalizationRecord(t, l, sigAggregrator, thirdBlock, nodes[0:quorum])
 
-	err = e.Start()
+	conf.Storage.Index(firstBlock, fCert1)
+	conf.Storage.Index(secondBlock, fCert2)
+	conf.Storage.Index(thirdBlock, fCer3)
+
+	e, err := NewEpoch(conf)
 	require.NoError(t, err)
+	require.Equal(t, uint64(3), e.Storage.Height())
+	require.NoError(t, e.Start())
 
+	// ensure the round is properly set to 3
 	require.Equal(t, uint64(3), e.Metadata().Round)
```

Comment:
can we add that the sequence is equal to 3 also? We had a bug with that, want to make sure when we recover we also compute it correctly.